core/src/test/java/hudson/util/SecretUtilTest.java

```diff
@@ -10,56 +10,32 @@
     @Test
     public void decrypt() {
         String data = "{}";
-
-        try {
-            Secret secret = Secret.decrypt(data);
-            Assert.assertNull(secret);
-        } catch (ArrayIndexOutOfBoundsException e) {
-            Assert.fail(data + " shouldn't throw an ArrayIndexOutOfBoundsException but returning null");
-        }
-
+        Secret secret = Secret.decrypt(data);
```

Comment:
This seems to lose some information about what we're trying to check for. Something like 
```suggestion
        Assert.assertNull(secret); // expected to not throw ArrayIndexOutOfBoundsException
```
or similar would be useful; no need read the linked issue then.

If deemed reasonable, applies to other changes as well.